builder/builder.go

```diff
@@ -220,9 +221,32 @@ func getWhereConditions(where map[string]interface{}) ([]Comparable, func(), err
 		return nil, emptyFunc, nil
 	}
 	wms := &whereMapSet{}
+	var comparables []Comparable
 	var field, operator string
 	var err error
 	for key, val := range where {
+		if key == "_or" {
+			var (
+				orWheres          []map[string]interface{}
+				orWhereComparable []OrNestWhere
+				ok                bool
+			)
+			if orWheres, ok = val.([]map[string]interface{}); !ok {
+				return nil, emptyFunc, errOrValueType
+			}
+			for _, orWhere := range orWheres {
+				if orWhere == nil {
+					continue
+				}
+				orNestWhere, f, err := getWhereConditions(orWhere)
+				defer f() // ?
```

Comment:
We're using a global variable `cpPool`, so I wonder if this line may doesn't work.

Reply:
I've been ran the golang race detector, and it seems no race detected (without your prompted changes):

```
➜  builder git:(build-select-or-operator) go test -race *.go

ok  	command-line-arguments	1.984s
```

I understand with your concerns, and I also agree that merge your `remove-pool` PR first, and then merge this PR after rebased master.

Thanks!